enhancements/deprecated-bundles.md

```diff
@@ -0,0 +1,131 @@
+
+#### Terminated bundle
+
+A terminated bundle is neither upgradable nor installable.
+
+### Goals
+
+- A deprecated bundle in the update graph and any of its replacements (terminated bundles) will not be installed by OLM
+- Operator Owners don’t need to release a new version in order to specify which bundle is deprecated
+- The index add command is enhanced to un-do this termination if needed
+- The deprecation is specified after promotion (not during) and before publishing of the index image
+- Alerts are fired on cluster when a deprecated version is installed
+
+### Non-Goals
+
+- OLM supports upgrading from a terminated bundle
+- Alerts are fired if terminated bundles exist on cluster
+- OLM prevents cluster upgrades based on whether versions are terminated in the upgraded cluster’s index
+
+## Proposal
+
+We propose to add a command to `opm index` called `deprecate` which takes in a set of bundle images, an index, and all the same arguments as `opm index add`. The command modifies the index by removing all versions that come before the specified bundle image in the update graph (that is, until the tail of each channel). If this operation removes the head of a channel, then the channel is removed. The removal of bundles from the update graph will not modify the update graph after the deprecated version (from the deprecated channel to the head of the channel).
+
+Versions that were removed from the index will not receive updates if subscriptions still exist for them on cluster. Ideally, appropriate communication from the operator maintainer will mean that cluster admins will upgrade to the latest supported version prior to cluster upgrade. In the event that the cluster is upgraded and `Subscriptions` to terminated bundles exist on cluster, the cluster admin can manually trigger an upgrade by either.
+
+- editing the `Subscription`'s `startingCSV` field (and/or `channel` field if the channel was removed) to point to an available / installable `csv`
+- creating a custom `CatalogSource` pointing to the previous OCP index, editing the `Subscription` to point to the new `CatalogSource` and, in the case where the channel was removed, editing the `channel` field of the `Subscription`.
+
+With the new work on the skip range being steppable, the above scenario may be less likely to happen.
+
+A new field will be added to the operator-registry database table `operatorbundle` called `deprecated` and the bundle images specified in the command will be marked as such.
+
+The OLM resolver will be modified with a new property type, (like `olm.deprecated`/`olm.obsolete`/`olm.not-installable`) in order to filter out any options that have this property set and disallow installs of deprecated bundles. This also means appropriate status reporting in the subscription and installplan need to be added to OLM, and alerts will be fired if a deprecated version is installed.
```

Comment:
let's define which here:

my vote is olm.deprecated to match with all the other names we are using for this command